File: Leecode/Array/414. 第三大的数/Solution.java
import java.util.Arrays;

class Solution {
    public int thirdMax(int[] nums) {

        nums = Arrays.sort(nums);
        int j = 0;
        for (int i = 0; i <= nums.length; i ++){
            if (i+2 > nums.length)
                if (j < 2)
                    return nums[0];
                else:
                    return nums[i];
            if (nums[i] != nums[i + 1] && j < 3)
                j ++;
            else if (nums[i+1] == null)
                return nums[i];
            if (j == 3)
                return nums[i]
        }
    }
}

File: Leecode/Array/414. 第三大的数/Solution.py
class Solution:
    def thirdMax(self, nums):
        nums = sorted(nums, reverse=True)
        print(nums)
        j = 0
        for i in range(len(nums)):
            if i+2 > len(nums) :
                if j < 2:
                    return nums[0]
                else:
                    return nums[i]
            if nums[i] != nums[i + 1] and j < 3:
                j += 1
            elif nums[i + 1] == None:
                return nums[i]
            if j == 3:
                return nums[i]
            print(str(j) + '~~~' + str(i))


num = [4, 3, 2, 2, 4, 3]
s1 = Solution()
s = s1.thirdMax(num)
print(s)